apps/rule-manager/client/src/ts/components/hooks/useTags.ts

```diff
@@ -8,11 +9,24 @@ export type Tag = {
   name: string
 }
 
+type TagRuleCount = {
+  tagId: number,
+  ruleCount: number
+}
+
+type TagRuleCounts = {
```

Comment:
My thinking was that the tag-rule count is only used on the tags page - I wouldn't want to create additional load elsewhere in the application (though it might admittedly be small) whenever `getTags` is called for information that isn't used in that context - especially as the tags table is very small and the tag-rule tables will be much larger.

Reply:
They're indexed, so I suspect the query performance for both is indistinguishable – but this'd be an interesting thing to prove! Appreciate it's a bit of a change, so we could ship this and refactor if we felt it was useful down the line.

Reply:
Ah, that makes sense - my intuition was more rows = more time taken, but if the indexing keeps it zippy then it would be good to avoid that extra fetch request. I'll give it a go!

Reply:
In case useful here or in future, this is the Postgres query analyser and visualiser I was mentioning @rhystmills that we found useful when considering query performance in Lurch: https://explain.dalibo.com/

If you add `EXPLAIN ANALYZE` followed by the query you're evaluating, you can then copy and paste the output into the visualiser and it'll break down the query for you, indicating which parts may be costly. E.g. https://explain.dalibo.com/plan/f653979ehdgh32e9#plan

Reply:
Thanks Samantha!

@jonathonherbert  - the list endpoint now includes tag-rule counts, and we no longer make a separate request for those.